File: CPP [3501-4000]/3798.largest_even_number [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    string largestEven(string s) {
        int idx = -1;
        for(int i=s.length()-1; i>=0; i--)
            if(s[i] == '2'){
                idx = i;
                break;
            }
        
        return s.substr(0, idx + 1);
    }
};

File: Javascript/3798.largest-even-number [1].js
/**
 * @param {string} s
 * @return {string}
 */
var largestEven = function(s) {
    let idx = -1
    for(let i=s.length-1; i>=0; i--){
        if(s[i] === '2'){
            idx = i
            break
        }
    }

    return s.substring(0, idx + 1)
};